eq-author-api/middleware/identification/index.js

```diff
@@ -0,0 +1,60 @@
+const verifyServiceRequest = require("./verifyServiceRequest");
+const { isNil, isEmpty } = require("lodash/fp");
+const jwt = require("jsonwebtoken");
+const { getUserByExternalId } = require("../../utils/datastore");
+
+module.exports = logger => async (req, res, next) => {
+  const authHeader = req.header(process.env.AUTH_HEADER_KEY || "authorization");
+  if (isNil(authHeader)) {
+    logger.error("Request must contain a valid authorization header.");
+    res.send(401);
+    return;
+  }
+
+  const accessToken = authHeader.replace("Bearer ", "").replace(/=/g, "");
+  if (isEmpty(accessToken)) {
+    logger.error("Request must contain a valid access token.");
+    res.send(401);
+    return;
+  }
+
+  const jwtToken = jwt.decode(accessToken);
+  if (isNil(jwtToken)) {
+    logger.error("Could not decode JWT token.");
+    res.send(401);
+    return;
+  }
+
+  if (jwtToken.serviceName) {
+    await verifyServiceRequest(accessToken, jwtToken.serviceName);
+    req.user = {
```

Comment:
Should this be `req.identity`?

Reply:
As discussed we will hold off the identity change until we have a discussion about what an identity should be.